app/uk/gov/hmrc/agentclientrelationships/services/CreateRelationshipsService.scala

```diff
@@ -57,53 +58,63 @@ class CreateRelationshipsService @Inject()(
     failIfAllocateAgentInESFails: Boolean)(
     implicit ec: ExecutionContext,
     hc: HeaderCarrier,
-    auditData: AuditData): Future[Option[Unit]] =
+    auditData: AuditData): Future[Option[DbUpdateStatus]] =
     lockService
       .tryLock(arn, identifier) {
 
         auditData.set("AgentDBRecord", false)
         auditData.set("enrolmentDelegated", false)
         auditData.set("etmpRelationshipCreated", false)
 
-        def createRelationshipRecord: Future[Unit] = {
+        def createRelationshipRecord: Future[DbUpdateStatus] = {
           val identifierType = TypeOfEnrolment(identifier).identifierKey
           val record = RelationshipCopyRecord(arn.value, identifier.value, identifierType, Some(oldReferences))
           relationshipCopyRepository
             .create(record)
-            .map(_ => auditData.set("AgentDBRecord", true))
+            .map(count => {
+              auditData.set("AgentDBRecord", true)
+              convertDbUpdateStatus(count)
+            })
             .recoverWith {
               case NonFatal(ex) =>
                 logger.warn(
                   s"Inserting relationship record into mongo failed for ${arn.value}, ${identifier.value} (${identifier.getClass.getSimpleName})",
                   ex)
                 if (failIfCreateRecordFails) Future.failed(new Exception("RELATIONSHIP_CREATE_FAILED_DB"))
-                else Future.successful(())
+                else Future.successful(DbUpdateFailed)
             }
         }
 
         for {
-          _ <- createRelationshipRecord
-          _ <- createEtmpRecord(arn, identifier)
-          _ <- createEsRecord(arn, identifier, failIfAllocateAgentInESFails)
-        } yield ()
+          recordCreationStatus <- createRelationshipRecord
+          if recordCreationStatus == DbUpdateSucceeded
+          etmpRecordCreationStatus <- createEtmpRecord(arn, identifier)
+          if etmpRecordCreationStatus == DbUpdateSucceeded
+          esRecordCreationStatus <- createEsRecord(arn, identifier, failIfAllocateAgentInESFails)
+        } yield esRecordCreationStatus
```

Comment:
I think this will fail with an exception if any of the predicates (== DbUpdateSucceeded) are not satisfied. Is this what you wanted or did you intend to return the failed state in case of failures? If so, you may need to add a `recover` that returns the failed state.

Reply:
There is a catch-all `recover` that now returns a 500 (SERVER ERROR)